client/lib/signup/step-actions.js

```diff
@@ -307,24 +301,12 @@ function processItemCart(
 	themeSlugWithRepo
 ) {
 	const addToCartAndProceed = () => {
-		let privacyItem = null;
-		const state = reduxStore.getState();
-		const { domainItem } = newCartItems;
-
-		if ( domainItem ) {
-			const { product_slug: productSlug } = domainItem;
-			const productsList = getProductsList( state );
-			if ( supportsPrivacyProtectionPurchase( productSlug, productsList ) ) {
-				privacyItem = updatePrivacyForDomain( domainItem, true );
-
-				if ( privacyItem ) {
-					newCartItems.push( privacyItem );
-				}
-			}
-		}
+		const newCartItemsToAdd = newCartItems.map( item =>
+			addPrivacyProtectionIfSupported( item, reduxStore )
```

Comment:
Why aren't you returning `addPrivacyProtectionIfSupported`?

Reply:
Oh I guess because `updatePrivacyForDomain` merges the privacy into the item.

Reply:
The `=>` syntax with the line split (by prettier) is a little deceptive--addPrivacyProtectionIfSupported is being returned.

Reply:
oh of course!